docs/PremiumApps.md

```diff
@@ -0,0 +1,369 @@
+
+The Splunk Operator installs the necessary Enterprise Security components depending on the architecture specified by the applied CRDs.
+
+#### Standalone Splunk Instance/ Standalone Search Heads
+For Standalone Splunk instances and standalone search heads the Operator will install Splunk Enterprise Security and all associated domain add-ons (DAs), and supporting add-ons (SAs).
+
+#### Search Head Cluster
+When installing Enterprise Security in a Search Head Cluster, the Operator will perform the following tasks: 
+1) Install the splunk enterprise app in Deployer's etc/apps directory.
+2) Run the ES post install command `essinstall` that stages the Splunk Enterprise Security and all associated domain add-ons (DAs) and supporting add-ons (SAs) to the etc/shcapps.
+3) Push the Search Head Cluster bundle from the deployer to all the SHs.
+
+#### Indexer Cluster
+When installing ES in an indexer clustering environment through the Splunk Operator it is necessary to deploy the supplemental [Splunk_TA_ForIndexers](https://docs.splunk.com/Documentation/ES/latest/Install/InstallTechnologyAdd-ons#Create_the_Splunk_TA_ForIndexers_and_manage_deployment_manually) app from the ES package to the indexer cluster members. This can be achieved using the AppFramework app deployment steps using appSources scope of "cluster".
+
+### How to Install Enterprise Security using the Splunk Operator
+
+#### Considerations for using the Splunk Operator
+
+When crafting your Custom Resource(CR) to create a Splunk Enterprise Deployment it is necessary to take the following configurations into account.
+
+##### [appSources](https://splunk.github.io/splunk-operator/AppFramework.html#appsources) scope
+   
+   - When deploying ES to a Standalone or to a Search Head Cluster, it must be configured with an appSources scope of "premiumApps".
+   - When deploying the Splunk_TA_ForIndexers app to an Indexer Cluster, it must be configured with an appSources scope of "cluster".
+
+#####  SSL Enablement
+
+When you install ES versions 6.3.0 or higher, you must supply a value for the parameter ssl_enablement that is required by the ES post installation command `essinstall`. By default, if you don't set any value for ssl_enablement, the value of `strict` is used which requires Splunk to have SSL enabled in web.conf (refer to setting `enableSplunkWebSSL`). The below table can be used for reference of available values of ssl enablement. 
+
+| SSL mode  | Description | 
+| --------- | ----------- | 
+|strict     | `Default mode`. Ensure that SSL is enabled in the web.conf configuration file to use this mode. Otherwise, the installer exists with an error. Note that for the SHC, the ES post install command `essinstall` is run on the deployer, and this command looks into web.conf files under etc/shcapps to validate that enableSplunkWebSSL is set to true or not. This is done with the assumption that you have already pushed a web.conf through etc/shcapps to all the SHC members.| 
```

Comment:
nit :it should be `....files under etc/shcluster/apps`

Reply:
good catch. fixed it.